### PR TITLE
Enforce attachment size limits at upload time.

### PR DESCRIPTION
**Pull request by jmcarp**
Enforce attachment size limit on uploading files to S3.

### CODE REVIEW

File: regulations/views/comment.py [hunk 1]
```diff
@@ -23,6 +31,7 @@ def upload_proxy(request):
     url = s3.generate_presigned_url(
         ClientMethod='put_object',
         Params={
+            'ContentLength': size,
```
Comment (cmc333333) [inline regulations/views/comment.py]: This is interesting. What happens if the uploaded file is actually larger than the content length provided? Will S3 trim the file? That'd be a pretty nice security mechanism.

Related, are there any special values we need to keep in mind? Do 0 or -1 mean "arbitrary size", for example?

Comment (jmcarp) [inline regulations/views/comment.py]: I didn't see documentation for this, but based on experimentation, but uploading a file that was too big errored with `SIGPIPE`, and uploading a file that was too small error with a signature mismatch. I'll check whether the file actually got uploaded in the first case.

And I don't know of any special values for `Content-Length`--if we don't want to validate, I think we just skip that parameter when generating the signed URL.

Comment (cmc333333) [inline regulations/views/comment.py]: If S3 errors when there's a content-length mismatch, we're golden. If it doesn't we might have a problem -- this file size data is coming from the UI (and therefore can't be trusted). Would you mind adding

```python
assert size > 0
```
just for good measure?

Comment (jmcarp) [inline regulations/views/comment.py]: Done. And I wasn't able to upload files that didn't match the content length in the signature.

**Comment by cmc333333** [comment]
Note that, before too long, we should start adding tests around all of this.

**Pull request merged**